v3/src/components/calculator/calculator.tsx

```diff
@@ -0,0 +1,106 @@
+import { Button, Flex, Text, useToast } from "@chakra-ui/react"
+import React, { useState } from "react"
+import { ITileBaseProps } from "../tiles/tile-base-props"
+import { isCalculatorModel } from "./calculator-model"
+
+import "./calculator.scss"
+
+export const CalculatorComponent = ({ tile }: ITileBaseProps) => {
+  const [calcValue, setCalcValue] = useState("")
+  const [justEvaled, setJustEvaled] = useState(false)
+  const toast = useToast()
+
+  const calculatorModel = tile?.content
+  if (!isCalculatorModel(calculatorModel)) return null
+
+  const clearValue = () => {
+    setCalcValue("")
+  }
+  const insertSymbol = (symbol: string) => { //parenthesis or decimal point
+    if (justEvaled) {
+      clearValue()
+    }
+    insert(symbol)
+  }
+  const insert = (strToInsert: string) => {
+    if (justEvaled && !isNaN(parseInt(strToInsert, 10))) {
+      clearValue()
+      setJustEvaled(false)
+      if (calcValue === "") { setCalcValue(strToInsert) }
+      else  { setCalcValue(calcValue.concat(strToInsert)) }
+    } else {
+      setCalcValue(calcValue.concat(strToInsert))
+    }
+  }
+  const evaluate = () => {
+    // eslint-disable-next-line no-useless-escape
+    const input = calcValue.replace(/[^0-9()+\-*\/.]/g, "") //sanitize input
+    if (calcValue.includes("(") && !calcValue.includes(")")) {
```

Comment:
This is one of a great number of possible syntax errors you've singled out. Is it part of a broader plan? Also noting that expressions like (8*9)+(7*8 are not reported with a toast but are still an error of missing a closing paren.
I guess I would recommend postponing any error handling beyond the try/catch until we have a general strategy.